tests/chainer_tests/functions_tests/array_tests/test_get_item.py

```diff
@@ -76,6 +76,118 @@ def test_backward_gpu(self):
                             cuda.to_gpu(self.gy_data))
 
 
+@testing.parameterize(*testing.product_dict(
+    [
+        {'slices': [], 'sliced_shape': (0, 3, 2)},
+        {'slices': [1, 0], 'sliced_shape': (2, 3, 2)},
+        {'slices': ([1, 0],), 'sliced_shape': (2, 3, 2)},
+        {'slices': numpy.array([[1, 0], [2, 3]]),
+         'sliced_shape': (2, 2, 3, 2)},
+        {'slices': ([1, 0], [1, 1]), 'sliced_shape': (2, 2)},
+        {'slices': ([1, 0], slice(None), [[1, 1], [1, 1]]),
+         'sliced_shape': (2, 2, 3)},
+        {'slices': ([1, 0], slice(1, 2), [0, 0]), 'sliced_shape': (2, 1)},
+        {'slices': ([[1, 1], [1, 0]], slice(1, 2), 1),
+         'sliced_shape': (2, 2, 1)},
+        {'slices': numpy.array([True] * 18 + [False] * 6).reshape(4, 3, 2),
+         'sliced_shape': (18,)},
+        {'slices': numpy.array([True, False, False, True]),
+         'sliced_shape': (2, 3, 2)},
+        {'slices': (slice(None), numpy.array([True, False, True])),
+         'sliced_shape': (4, 2, 2)},
+    ],
+    [
+        {'dtype': numpy.float32},
+        {'dtype': numpy.int32},
+        {'dtype': numpy.uint32},
+        {'dtype': numpy.uint64},
+        {'dtype': numpy.ulonglong},
+    ]
+))
+class TestGetitemAdvanced(unittest.TestCase):
+
+    def setUp(self):
+        self.x_data = numpy.random.uniform(
+            -1, 1, (4, 3, 2)).astype(numpy.float32)
+        self.gy_data = numpy.random.uniform(
+            -1, 1, self.sliced_shape).astype(numpy.float32)
+
+    def check_forward(self, x_data):
+        x = chainer.Variable(x_data)
+        y = functions.get_item(x, self.slices)
+        self.assertEqual(y.data.dtype, numpy.float32)
+        numpy.testing.assert_equal(cuda.to_cpu(x_data)[self.slices],
+                                   cuda.to_cpu(y.data))
+
+    def test_forward_cpu(self):
+        self.check_forward(self.x_data)
+
+    @attr.gpu
+    def test_forward_gpu(self):
+        self.check_forward(cuda.to_gpu(self.x_data))
+
+    def check_backward(self, x_data, y_grad):
+        gradient_check.check_backward(functions.GetItem(self.slices),
+                                      (x_data,), y_grad)
+
+    def test_backward_cpu(self):
+        self.check_backward(self.x_data, self.gy_data)
+
+    @attr.gpu
+    def test_backward_gpu(self):
+        self.check_backward(cuda.to_gpu(self.x_data),
+                            cuda.to_gpu(self.gy_data))
+
+
+@parameterize(
+    {'slices': ([1, 0], [1, 1]), 'sliced_shape': (2, 2)},
+    {'slices': ([1, 0], slice(None), [[1, 1], [1, 1]]),
+     'sliced_shape': (2, 2, 3)},
+    {'slices': ([1, 0], [1, 1], [0, 0]), 'sliced_shape': (2,)},
+    {'slices': (slice(None), numpy.array([True, False, True])),
+     'sliced_shape': (4, 2, 2)},
+)
```

Comment:
My understanding is:
- `TestCupyIndicesGetItem` is to test advanced indexing when `slices` contains `cupy.ndarray`.
- Since you cannot set `cupy.ndarray` directly inside `@parameterize`, you set `numpy.ndarray` and convert it to `cupy.ndarray` inside `@attr.gpu`.

Is that right? I still don't understand why here you also include test cases not containing `numpy.ndarray`, which seem to be already tested in `TestGetItemAdvanced`.